lib/puppet/type/file/source.rb

```diff
@@ -352,5 +352,12 @@ def chunk_file_from_source
 
     defaultto :ignore
     newvalues(:use, :use_when_creating, :ignore)
+     munge do |value|
+      if @resource.file && @resource.line && value != :ignore
```

Comment:
This needs to be `value.to_s != 'ignore'`. When `source_permissions` appears in a manifest `value` will be a String. When `source_permissions` is used programmatically, like the downloader it's a symbol. The munge method needs to normalize it to one of the enumerated values. So the check here needs to handle both, otherwise you get the warning when it's set to `ignore` in a manifest:

```
# puppet agent -t --server hhd1w8tsjjcgdq2.delivery.puppetlabs.net
SOURCE_PERMISSION: :ignore
SOURCE_PERMISSION: :ignore
SOURCE_PERMISSION: :ignore
SOURCE_PERMISSION: :ignore
SOURCE_PERMISSION: :ignore
SOURCE_PERMISSION: :ignore
SOURCE_PERMISSION: :ignore
SOURCE_PERMISSION: :ignore
SOURCE_PERMISSION: :ignore
SOURCE_PERMISSION: :ignore
SOURCE_PERMISSION: :ignore
SOURCE_PERMISSION: :ignore
SOURCE_PERMISSION: :ignore
SOURCE_PERMISSION: :ignore
SOURCE_PERMISSION: :ignore
SOURCE_PERMISSION: :ignore
SOURCE_PERMISSION: :ignore
SOURCE_PERMISSION: :ignore
SOURCE_PERMISSION: :ignore
SOURCE_PERMISSION: :ignore
SOURCE_PERMISSION: :ignore
SOURCE_PERMISSION: :ignore
SOURCE_PERMISSION: :ignore
SOURCE_PERMISSION: :ignore
SOURCE_PERMISSION: :ignore
SOURCE_PERMISSION: :ignore
SOURCE_PERMISSION: :ignore
SOURCE_PERMISSION: :ignore
SOURCE_PERMISSION: :ignore
SOURCE_PERMISSION: :ignore
SOURCE_PERMISSION: :ignore
SOURCE_PERMISSION: :ignore
SOURCE_PERMISSION: :ignore
Info: Using configured environment 'production'
Info: Retrieving pluginfacts
SOURCE_PERMISSION: :use
Info: Retrieving plugin
SOURCE_PERMISSION: :ignore
Info: Retrieving locales
SOURCE_PERMISSION: :ignore
Info: Caching catalog for hhd1w8tsjjcgdq2.delivery.puppetlabs.net
SOURCE_PERMISSION: "ignore"
Warning: The `source_permissions` parameter is deprecated. Explicitly set `owner`, `group`, and `mode`.
   (file: /etc/puppetlabs/code/environments/production/manifests/site.pp, line: 2)
Info: Applying configuration version '1533940355'
Notice: Applied catalog in 0.02 seconds
```